src/anet.c

```diff
@@ -546,17 +546,23 @@ int anetFdToString(int fd, char *ip, size_t ip_len, int *port, int fd_to_str_typ
     } else {
         if (getsockname(fd, (struct sockaddr *)&sa, &salen) == -1) goto error;
     }
-    if (ip_len == 0) goto error;
 
     if (sa.ss_family == AF_INET) {
         struct sockaddr_in *s = (struct sockaddr_in *)&sa;
-        if (ip) inet_ntop(AF_INET,(void*)&(s->sin_addr),ip,ip_len);
+        if (ip) {
+            if (inet_ntop(AF_INET,(void*)&(s->sin_addr),ip,ip_len) == NULL)
+                goto error;
+        }
         if (port) *port = ntohs(s->sin_port);
     } else if (sa.ss_family == AF_INET6) {
         struct sockaddr_in6 *s = (struct sockaddr_in6 *)&sa;
-        if (ip) inet_ntop(AF_INET6,(void*)&(s->sin6_addr),ip,ip_len);
+        if (ip) {
+            if (inet_ntop(AF_INET6,(void*)&(s->sin6_addr),ip,ip_len) == NULL)
+                goto error;
+        }
         if (port) *port = ntohs(s->sin6_port);
     } else if (sa.ss_family == AF_UNIX) {
+        if (ip_len < sizeof("/unixsocket")) goto error;
         if (ip) snprintf(ip, ip_len, "/unixsocket");
```

Comment:
I think this should be something like this, this way we don't duplicate `"/unixsocket"` and handle the `ip_len` check implicitly as parse of `snprintf`.
```suggestion
        if (ip) {
            int res = snprintf(ip, ip_len, "/unixsocket");
            if (res < 0 || res >= ip_len) goto error;
        }
```

Reply:
Thanks, I was in a hurry to fix the compile error. This is much better.